message_ix_models/model/material/build.py

```diff
@@ -65,98 +68,55 @@
     "power_sector",
     "fertilizer",
     "methanol",
-]
+)
 
 
 # Try to handle multiple data input functions from different materials
-def add_data_1(scenario, dry_run=False):
+def add_data(scenario, dry_run=False):
     """Populate `scenario` with MESSAGEix-Materials data."""
     # Information about `scenario`
-    info = ScenarioInfo(scenario)
-
-    # Check for two "node" values for global data, e.g. in
-    # ixmp://ene-ixmp/CD_Links_SSP2_v2.1_clean/baseline
-    if {"World", "R11_GLB"} < set(info.set["node"]):
-        log.warning("Remove 'R11_GLB' from node list for data generation")
-        info.set["node"].remove("R11_GLB")
-    if {"World", "R12_GLB"} < set(info.set["node"]):
-        log.warning("Remove 'R12_GLB' from node list for data generation")
-        info.set["node"].remove("R12_GLB")
-
-    for func in DATA_FUNCTIONS_1 + DATA_FUNCTIONS_2:
+    for func in DATA_FUNCTIONS:
         # Generate or load the data; add to the Scenario
         log.info(f"from {func.__name__}()")
         data = func(scenario)
-        # if "SSP_dev" in scenario.model:
-        #     if "emission_factor" in list(data.keys()):
-        #         data.pop("emission_factor")
+        data = {k: v for k, v in data.items() if not v.empty}
         add_par_data(scenario, data, dry_run=dry_run)
-
-    log.info("done")
-
-
-def add_data_2(scenario, dry_run=False):
-    """Populate `scenario` with MESSAGEix-Materials data."""
-    # Information about `scenario`
-    info = ScenarioInfo(scenario)
-
-    # Check for two "node" values for global data, e.g. in
-    # ixmp://ene-ixmp/CD_Links_SSP2_v2.1_clean/baseline
-    if {"World", "R11_GLB"} < set(info.set["node"]):
-        log.warning("Remove 'R11_GLB' from node list for data generation")
-        info.set["node"].remove("R11_GLB")
-    if {"World", "R12_GLB"} < set(info.set["node"]):
-        log.warning("Remove 'R12_GLB' from node list for data generation")
-        info.set["node"].remove("R12_GLB")
-
-    for func in DATA_FUNCTIONS_2:
-        # Generate or load the data; add to the Scenario
-        log.info(f"from {func.__name__}()")
-        # TODO: remove this once emission_factors are back in SSP_dev
-        data = func(scenario)
-        # if "SSP_dev" in scenario.model:
-        #     if "emission_factor" in list(data.keys()):
-        #         data.pop("emission_factor")
-        add_par_data(scenario, data, dry_run=dry_run)
-
     log.info("done")
 
 
 def build(
-    scenario: message_ix.Scenario, old_calib: bool, iea_data_path=None
+    context: Context,
+    scenario: message_ix.Scenario,
+    old_calib: bool,
+    modify_existing_constraints=True,
+    iea_data_path=None,
```

Comment:
```suggestion
    iea_data_path: Optional[Path] = None,
```

You could also add type hints here: to enable this one (if it's supposed to be a path), you might need to `from pathlib import Path` and `from typing import Optional` first. If it's just a string, please indicate that, too.

Fun fact: starting with Python 3.10, you can use the much more readible `iea_data_path: Path | None = None`.